From 830c5daef062232d899a839415d27b350bf43fcc Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Dec 2024 16:14:17 +0800
Subject: [PATCH 195/291] AOSCOS: platform: lemote3a-laptop: drop fb_blank
 state from backlight restoration logic

Follow commit 4551978bb50a ("backlight: Remove fb_blank from struct
backlight_properties") and remove fb_blank from the ternary operator in
backlight restoration function, as the unblank state has already been set
as the default since this commit.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/platform/mips/lemote3a-laptop.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/platform/mips/lemote3a-laptop.c b/drivers/platform/mips/lemote3a-laptop.c
index b06e7b6ae97c..6e6b0c46aca4 100644
--- a/drivers/platform/mips/lemote3a-laptop.c
+++ b/drivers/platform/mips/lemote3a-laptop.c
@@ -625,9 +625,8 @@ static int lemote3a_set_brightness(struct backlight_device * pdev)
 {
 	unsigned int level = 0;
 
-	level = ((FB_BLANK_UNBLANK==pdev->props.fb_blank) &&
-				(FB_BLANK_UNBLANK==pdev->props.power)) ?
-					pdev->props.brightness : 0;
+	level = (FB_BLANK_UNBLANK==pdev->props.power) ?
+				pdev->props.brightness : 0;
 
 	if (MAX_BRIGHTNESS < level) {
 		level = MAX_BRIGHTNESS;
-- 
2.48.1

